cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,201 @@
+            + "|.*-909[1-4]\\.ssl\\.truststore\\.password"
+            + "|.*-909[1-4]\\.ssl\\.truststore\\.type"
+            + "|.*-909[1-4]\\.ssl\\.client\\.auth"
+            + "|.*-909[1-4]\\.scram-sha-512\\.sasl\\.jaas\\.config"
+            + "|.*-909[1-4]\\.sasl\\.enabled\\.mechanisms"
+            + "|advertised\\.listeners"
+            + "|zookeeper\\.connect"
+            //+ "|log\\.dirs"*/
+            + "|super\\.users"
+            + "|broker\\.rack)$");
+
+    public KafkaBrokerConfigurationDiff(Map<ConfigResource, Config> current, ConfigMap desired, KafkaVersion kafkaVersion, int brokerId) {
+        this.current = current;
+        this.diff = emptyKafkaConf(); // init
+        this.desired = desired;
+        this.kafkaVersion = kafkaVersion;
+        this.brokerId = brokerId;
+        this.deletedEntries = new ArrayList<String>();
+        Config brokerConfigs = current.get(new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(brokerId)));
+        if (brokerConfigs == null) {
+            throw new RuntimeException("Failed to get broker " + brokerId + " configuration");
+        }
+        this.currentEntries = brokerConfigs.entries();
+        this.diff = computeDiff();
+    }
+
+    private KafkaConfiguration emptyKafkaConf() {
+        Map<String, Object> conf = new HashMap<>();
+        return new KafkaConfiguration(conf.entrySet());
+    }
+
+    private HashMap<String, String> configMap2Map() {
+        HashMap<String, String> result = new HashMap<String, String>();
+        if (desired == null || desired.getData() == null || desired.getData().get("server.config") == null) {
+            return result;
+        }
+        String desireConf = desired.getData().get("server.config");
+
+        List<String> list = getLinesWithoutCommentsAndEmptyLines(desireConf);
+        for (String line: list) {
+            String[] split = line.split("=");
+            if (split.length == 1) {
+                result.put(split[0], "");
+            } else {
+                result.put(split[0], split[1]);
+            }
+        }
+        return result;
+    }
+
+    private List<String> getLinesWithoutCommentsAndEmptyLines(String config) {
+        List<String> allLines = Arrays.asList(config.split("\\r?\\n"));
+        List<String> validLines = new ArrayList<>();
+
+        for (String line : allLines)    {
+            if (!line.replace(" ", "").startsWith("#") && !line.isEmpty())   {
+                validLines.add(line.replace(" ", ""));
+            }
+        }
+
+        return validLines;
+    }
+
+    private void fillPlaceholderValue(Map<String, String> map, String placeholder, String value) {
+        map.entrySet().forEach(entry -> {
+            if (!IGNORABLE_PROPERTIES.matcher(entry.getKey()).matches() && entry.getValue().contains("${" + placeholder + "}")) {
+                entry.setValue(entry.getValue().replaceAll("\\$\\{" + placeholder + "\\}", value));
```

Comment:
I guess you need to quote `placeholder` because it could contain regex metacharacters (e.g. `.`).